packages/ag-charts-enterprise/src/features/sync/chartSync.ts

```diff
@@ -0,0 +1,106 @@
+import { type AgChartSyncOptions, _ModuleSupport } from 'ag-charts-community';
+
+const { BOOLEAN, STRING, UNION, BaseProperties, ChartUpdateType, ObserveChanges, Validate } = _ModuleSupport;
+
+export class ChartSync extends BaseProperties implements _ModuleSupport.ModuleInstance, AgChartSyncOptions {
+    static className = 'Sync';
+
+    @Validate(BOOLEAN)
+    enabled: boolean = false;
+
+    @Validate(STRING, { optional: true })
+    @ObserveChanges<ChartSync>((target, newValue, oldValue) => {
+        const { syncManager } = target.moduleContext;
+        syncManager.unsubscribe(oldValue);
+        syncManager.subscribe(newValue);
+        target.updateSiblings(oldValue);
+        target.updateSiblings(newValue);
+    })
+    groupId?: string;
+
+    @Validate(UNION(['x', 'y', 'xy'], 'an axis'))
+    @ObserveChanges<ChartSync>((target) => {
+        const { syncManager } = target.moduleContext;
+        target.updateChart(syncManager.getChart());
+    })
+    axes: 'x' | 'y' | 'xy' = 'x';
+
+    @Validate(BOOLEAN)
+    @ObserveChanges<ChartSync>((target, newValue) => {
+        if (newValue) {
+            target.enabledZoomSync();
+        } else {
+            target.disableZoomSync?.();
+        }
+    })
+    zoom: boolean = true;
+
+    constructor(protected moduleContext: _ModuleSupport.ModuleContext) {
+        super();
+    }
+
+    private updateChart(chart: any) {
+        chart.updateService.update(ChartUpdateType.UPDATE_DATA, { skipSync: true });
+    }
+
+    private updateSiblings(groupId?: string) {
+        for (const chart of this.moduleContext.syncManager.getSiblings(groupId)) {
+            this.updateChart(chart);
+        }
+    }
+
+    private enabledZoomSync() {
+        const { syncManager, zoomManager } = this.moduleContext;
+        this.disableZoomSync = zoomManager.addListener('zoom-change', (e) => {
+            if ('stopPropagation' in e && e.stopPropagation) return;
+            for (const chart of syncManager.getSiblings(this.groupId)) {
+                chart.zoomManager.updateZoom('zoom', zoomManager.getZoom());
+                (chart.zoomManager as any).listeners.dispatch('zoom-change', { ...e, stopPropagation: true });
+            }
+        });
+    }
+
+    private disableZoomSync?: () => void;
+
+    syncAxes(stopPropagation = false) {
+        const { syncManager } = this.moduleContext;
+        const chart = syncManager.getChart();
+        const syncSeries = syncManager.getGroup(this.groupId).flatMap((chart) => chart.series);
+
+        for (const axis of chart.axes) {
+            if (this.axes !== 'xy' && this.axes !== axis.direction) continue;
+            axis.boundSeries = syncSeries.filter((series) => {
+                const seriesKeys = series.getKeys(axis.direction);
+                return axis.keys.length ? axis.keys.some((key) => seriesKeys?.includes(key)) : true;
+            });
```

Comment:
I have some concerns about the scalability of this, every chart axis instance has references to every other charts series?

Reply:
It is currently only used for calculating the domain, and there's nothing more to it.
I agree it's not ideal but I believe the solution is a refactor of the axis and chart implementation.

Reply:
Discussed offline, we should follow up with a refactor to try and decouple axis & scales, so we could avoid O(n^3) complexity in the domain calculation.